src/main/java/rx/Observable.java

```diff
@@ -7771,7 +7771,10 @@ public final Subscription subscribe(Subscriber<? super T> subscriber) {
      * @see <a href="http://reactivex.io/documentation/operators/takelast.html">ReactiveX operators documentation: TakeLast</a>
      */
     public final Observable<T> takeLast(final int count) {
-        return lift(new OperatorTakeLast<T>(count));
+        if (count == 1 )
```

Comment:
Is takeLast(0) meaningful? If so, it can be routed to ```ignoreElements()```. BTW, ```ignoreElements()``` can be optimized by making it singleton and be very simple (i.e., its onNext does nothing) instead of filtering with always false.